### PR TITLE
meson: create dist archives suitable for building with configure, too

### PR DESCRIPTION
**Pull request by eli-schwartz**
Run autogen.sh as a dist script while creating the tarball. Also update autogen.sh to detect when it is being run from `meson dist`, and use that as the srcdir.

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -23,6 +23,13 @@
 
 PROJECT="libva"
 
+# for `meson dist`
+if test -z "$srcdir"; then
+    srcdir="$MESON_PROJECT_DIST_ROOT"
+    test -n "$srcdir" || srcdir="$MESON_DIST_ROOT"
+    NOCONFIGURE=1
```
Comment (dvrogozh) [inline autogen.sh:30]: As caught by ci, this line breaks typically autotools build flow many people have:
```
$ ./autogen.sh
<...>
$ make
make: *** No targets specified and no makefile found.  Stop.
```

Can we test if we under meson upfront checking existence of some meson specific environment variable and avoid breaking autotools experience? Maybe:
```
if test -n "$MESON_PROJECT_DIST_ROOT"
<...>
fi
```

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: autogen.sh (hunk 1 above)
Comment (eli-schwartz) [reply autogen.sh]: We can check if the previous srcdir= assignment succeeded, I guess. ;)

Comment (eli-schwartz) [reply autogen.sh]: I will add that the autotools flow many people have is broken and wrong and utterly incorrect -- people should never ever do this under any circumstances.

This is of course a personal opinion and I didn't intend to break it.

Comment (dvrogozh) [reply autogen.sh]: > We can check if the previous srcdir= assignment succeeded, I guess. ;)

Works for me.

**Review by dvrogozh — APPROVED** [review]
looks good. @XinfengZhang : any objections to let this merged?

**Review by XinfengZhang — APPROVED** [review]

**Pull request merged**